x/wasm/keeper/msg_server_integration_test.go

```diff
@@ -1144,3 +1144,73 @@ func TestStoreAndMigrateContract(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateContractLabel(t *testing.T) {
+	wasmApp := app.Setup(t)
+	ctx := wasmApp.BaseApp.NewContextLegacy(false, tmproto.Header{Time: time.Now()})
+
+	var (
+		myAddress       sdk.AccAddress = make([]byte, types.ContractAddrLen)
+		authority                      = wasmApp.WasmKeeper.GetAuthority()
+		_, _, otherAddr                = testdata.KeyTestPubAddr()
+	)
+
+	specs := map[string]struct {
+		addr   string
+		expErr bool
+	}{
+		"authority can update contract label": {
+			addr:   authority,
+			expErr: false,
+		},
+		"admin can update contract label": {
+			addr:   myAddress.String(),
+			expErr: false,
+		},
+		"other address cannot update contract label": {
```

Comment:
🤔 good to have the scenarios covered but there is some overlap with the keeper tests. 
The integration tests give more confidence on the whole process. If you see value in both, then we can keep them.